prover/src/aggregator/prover.rs

```diff
@@ -17,20 +18,26 @@ use crate::{
 };
 
 #[derive(Debug)]
-pub struct Prover {
+pub struct Prover<'params> {
     // Make it public for testing with inner functions (unnecessary for FFI).
-    pub prover_impl: common::Prover,
+    pub prover_impl: common::Prover<'params>,
     pub chunk_protocol: Vec<u8>,
     raw_vk_batch: Option<Vec<u8>>,
     raw_vk_bundle: Option<Vec<u8>>,
 }
 
-impl Prover {
-    pub fn from_dirs(params_dir: &str, assets_dir: &str) -> Self {
+impl<'params> Prover<'params> {
+    pub fn degrees() -> Vec<u32> {
```

Comment:
we don't need this `degrees` method in both the aggregetor/zkevm verifier, since we don't use it. The degrees needed to instantiate the `Prover` are more than the degrees needed to instantiate the `Verifier` (which is used by `libzkp`). Since that part is already handled by simply the `layer2_degree` and `layer4_degree`, we don't need to expose another function `degrees()` here. So I am removing it.